pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -707,6 +708,8 @@ func manageServiceAccountCABundle(ctx context.Context, lister corev1listers.Conf
 	requiredConfigMap, err := resourcesynccontroller.CombineCABundleConfigMaps(
 		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.TargetNamespace, Name: "serviceaccount-ca"},
 		lister,
+		"apiserver-auth",
```

Comment:
this doesn't look right.  The producer and consumer here is the kube-controller-manager.  I don't think this can simply be assigned to the auth team.  There is *some* claim that the kube-apiserver team owns connections to the kube-apiserver and I'd let that play out between @ingvagabund and @tkashem.

cc @stlaz